### ISSUE TITLE
upper_case_acronyms contains about full-capital identifiers

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang/rust-clippy#6631]
Lint name: upper_case_acronyms

I tried this code:

```rust
pub enum DIR {}
```
`Checks for camel case name containing a capitalized acronym.` but there is no camel-case here, it's literally all uppercase, not sure why this triggers here then....

**Comment by mikerite** [comment]
There is no way the lint can tell if you're writing an acronym in camel case or one word in uppercase. According to the API guidelines, enum names should be in "UpperCamelCase" and the lint assumes that you're sticking to those guidelines.

**Comment by matthiaskrgr** [comment]
If the lint can't be sure that something is a TP and not a FP, I would rather have it not emit anything because FPs are generally more annoying than FNs.

**Comment by mikerite** [comment]
This lint is about enforcing a rule in the [naming guidelines](https://rust-lang.github.io/api-guidelines/naming.html#casing-conforms-to-rfc-430-c-case). It's enabled by default because projects should follow the guidelines. Projects that are using their own naming standard (for some reason) should just allow this lint.

That warning is correct in every project that follows the Rust style convention that enums are in UpperCamelCase.

**Comment by camsteffen** [comment]
The lint is right, but it's reasoning is wrong. The lint name and/or message could be improved so that there is no assumption that "DIR" is an acronym. It could instead say

> Names should be camel-cased and acronyms should not be capitalized after the first letter.

**Comment by skondrashov** [comment]
Is it clippy's responsibility to enforce these kinds of formatting guidelines? I don't see any other rules concerning camel or snake case enforcement, and I would expect this to be a rustfmt warning in similar situations (eg "structure field `FIELD` should have a snake case name" for struct fields). I am getting this warning for enum variants, which is not only confusing since I am also not using acronyms, but I think also wrong because enum variants can be considered to be constants, which is why I am ALLCAPSing mine. It doesn't seem right to be encouraged to rewrite enums as a list of constants to dodge this rule.

**Comment by skondrashov** [comment]
Sorry, naming - but that's my question, is it clippy's responsibility to enforce naming conventions if it's usually rustc's responsibility?

**Comment by ankddev** [comment]
Hello! I wanted to fix this one, but for me it doesn't give any errors:

```
$ cat src/main.rs
─────┬─────────────────────────────
     │ File: src/main.rs
─────┼─────────────────────────────
   1 │ pub enum DIR {}
   2 │
   3 │ fn main() {}
─────┴─────────────────────────────
$ cargo clippy -V
clippy 0.1.94 (4a4ef493e3 2026-03-02)
$ cargo -Vv
cargo 1.94.0 (85eff7c80 2026-01-15)
release: 1.94.0
commit-hash: 85eff7c80277b57f78b11e28d14154ab12fcf643
commit-date: 2026-01-15
host: x86_64-pc-windows-gnu
libgit2: 1.9.2 (sys:0.20.3 vendored)
libcurl: 8.17.0-DEV (sys:0.4.84+curl-8.17.0 vendored ssl:Schannel)
os: Windows 10.0.19045 (Windows 10 Pro) [64-bit]
$ cargo clippy -- -D clippy::upper_case_acronyms
    Checking test_rust v0.1.0 (C:\Users\user\projects\test_rust)
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.22s
$
```

Am i doing something wrong? I wasn't able to find it in the documentation as well.

**Comment by Zhonghe-zhao** [comment]
Sorry, I won't be able to work on this issue anymore. Could you please unassign me? Thanks!
